### PR TITLE
Migrate VisuallyHidden component

### PR DESCRIPTION
**Pull request by dougmacknz**
PR title says it all

### CODE REVIEW

File: packages/components/src/VisuallyHidden/VisuallyHidden.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import styles from "./VisuallyHidden.module.scss"
+
+export type AllowedTags = "div" | "span"
+
+export type VisuallyHiddenProps = {
+  children: ReactNode
+  /**
+   * The HTML tag rendered by this component
+   */
+  tag?: AllowedTags
```
Comment (gyfchong) [inline packages/components/src/VisuallyHidden/VisuallyHidden.tsx:12]: Would be good to have a short description in the docs describing what this does.

File: packages/components/src/VisuallyHidden/VisuallyHidden.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  it("renders a `data-testid` attribute", () => {
+    const { getByTestId } = render(
+      <VisuallyHidden data-testid="test-id">
+        <p>Lorem ipsum</p>
+      </VisuallyHidden>
+    )
+
+    expect(getByTestId("test-id")).toBeTruthy()
+  })
```
Comment (gyfchong) [inline packages/components/src/VisuallyHidden/VisuallyHidden.spec.tsx:25]: We decided somewhere that this is a pretty useless test. Remove

File: packages/components/src/VisuallyHidden/VisuallyHidden.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+} & OverrideClassName<HTMLAttributes<HTMLDivElement>>
+
+export const VisuallyHidden = ({
+  children,
+  classNameOverride,
+  tag = "span",
+  ...otherProps
+}: VisuallyHiddenProps): JSX.Element => {
+  const className = `${styles.srOnly} ${classNameOverride || ""}`
```
Comment (HeartSquared) [inline packages/components/src/VisuallyHidden/VisuallyHidden.tsx:21]:
```suggestion
  const className = classnames(styles.srOnly, classNameOverride)
```

File: packages/components/src/VisuallyHidden/VisuallyHidden.tsx (hunk 1 above)
Comment (HeartSquared) [inline packages/components/src/VisuallyHidden/VisuallyHidden.tsx:12]: Is there any reason this would even be anything other than a `span`? 🤔

Comment (dougmacknz) [reply packages/components/src/VisuallyHidden/VisuallyHidden.tsx:12]: Good question. I can't see any usages in our repos that are using this prop. I'll remove it

File: packages/components/src/VisuallyHidden/VisuallyHidden.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+describe("<VisuallyHidden />", () => {
+  it("still renders any child components", () => {
+    const { getByText } = render(
+      <VisuallyHidden>
+        <p>Lorem ipsum</p>
+      </VisuallyHidden>
+    )
+
+    expect(getByText("Lorem ipsum")).toBeTruthy()
```
Comment (HeartSquared) [inline packages/components/src/VisuallyHidden/VisuallyHidden.spec.tsx:14]:
```suggestion
    expect(getByText("Lorem ipsum")).toBeInTheDocument()
```

File: packages/components/src/VisuallyHidden/VisuallyHidden.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  children: ReactNode
+} & OverrideClassName<HTMLAttributes<HTMLDivElement>>
+
+export const VisuallyHidden = ({
+  children,
+  classNameOverride,
+  ...otherProps
+}: VisuallyHiddenProps): JSX.Element => {
+  const className = classnames(styles.srOnly, classNameOverride)
+  return createElement("span", { ...otherProps, className }, children)
```
Comment (HeartSquared) [inline packages/components/src/VisuallyHidden/VisuallyHidden.tsx:16]: Much easier to read :D
```suggestion
  <span
    classnames(styles.srOnly, classNameOverride)
    {...restProps}
  >
    {children}
  </span>
```

**Review by HeartSquared — APPROVED** [review]

**Pull request merged**